pkg/operator/controller/externaldns/deployment_test.go

```diff
@@ -5002,6 +5039,96 @@ func TestSecurityContextChanged(t *testing.T) {
 	}
 }
 
+func TestDeploymentStrategyChanged(t *testing.T) {
+	twentyPercent := intstr.FromString("20%")
+
+	for _, tc := range []struct {
+		name    string
+		current *appsv1.Deployment
+		desired *appsv1.Deployment
+		changed bool
+	}{
+		{
+			name:    "Recreate added for first time",
+			current: &appsv1.Deployment{},
+			desired: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Strategy: appsv1.DeploymentStrategy{
+						Type: "Recreate",
+					},
+				},
+			},
+			changed: true,
+		},
+		{
+			name: "Changed from non empty RollingUpdate to Recreate",
+			current: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Strategy: appsv1.DeploymentStrategy{
+						Type: "RollingUpdate",
+						RollingUpdate: &appsv1.RollingUpdateDeployment{
+							MaxUnavailable: &twentyPercent,
+							MaxSurge:       &twentyPercent,
+						},
+					},
+				},
+			},
+			desired: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Strategy: appsv1.DeploymentStrategy{
+						Type: "Recreate",
+					},
+				},
+			},
+			changed: true,
+		},
+		{
+			name: "No changes when Recreate is expected",
+			current: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Strategy: appsv1.DeploymentStrategy{
+						Type: "Recreate",
+					},
+				},
+			},
+			desired: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Strategy: appsv1.DeploymentStrategy{
+						Type: "Recreate",
+					},
+				},
+			},
+			changed: false,
+		},
+		{
+			name: "No changes when nothing is expected",
+			current: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Strategy: appsv1.DeploymentStrategy{
+						Type: "Recreate",
+					},
+				},
+			},
+			desired: &appsv1.Deployment{},
+			changed: false,
+		},
```

Comment:
You need a test case here where you try to update it from Recreate to RollingUpdate, and it should not change.

Reply:
What real life case does this cover? A developer who changes the desired strategy from Rolling to Recreate?

Reply:
No, a change from Recreate to Rolling.

Reply:
If the user changes from recreate to rolling - this case is covered in `Changed from non empty RollingUpdate to Recreate` test.
If the developer changes the desired from the recreate to rolling - there must be a lot more code to be done including the test cases. The code I dropped after your suggestion to not allow the rolling strategy was essentially doing this.

Reply:
I am asking about a `Changed from Recreate to RollingUpdate should fail` test.  `from` is the previous state and `to` is the attempted updated state.

Reply:
But this assumes that `Rolling` is a valid **desired** strategy which we already said we didn't want.

Reply:
In unit tests, we test invalid stuff to make sure it doesn't work.

Reply:
I'm sorry but I don't quite see how this test case should look like and how the reconciliation is supposed to fail. What exactly should `externalDNSDeploymentStrategyChanged` function do and return if the expected strategy has been set as`Rolling`?

Reply:
If the expected strategy is set to `Rolling` and we want to enforce the `Recreate` strategy, then you set it to `Recreate` and validate that.  I can't answer the second question, but it doesn't mean it isn't possible to do so.

Reply:
Miciah mentioned a better way to test an attempted change from Recreate to RollingUpdate would be in a different test function, the TestDesiredExternalDNSDeployment.  That would be fine.